treatment-service/config/config.go

```diff
@@ -37,7 +37,7 @@ type Config struct {
 
 type AssignedTreatmentLoggerConfig struct {
 	Kind                 AssignedTreatmentLoggerKind `default:""`
-	QueueLength          int                         `default:"1073741824"`
```

Comment:
Not sure why the default value is `1073741824` (which coincidentally coincides with the size of 1 gibibyte), but basically `QueueLength` is used for creating a [buffered channel](https://github.com/gojek/turing-experiments/blob/main/treatment-service/monitoring/treatmentlogger.go#L99) to receive and publish `AssignedTreatmentLog`.

By adding more logs to debug the issue, these failed tests pin-point the failing line to where the buffered channel is created.

`Prior to using a lower channel size`
https://github.com/gojek/turing-experiments/runs/7515128336?check_suite_focus=true
https://github.com/gojek/turing-experiments/runs/7514891584?check_suite_focus=true
https://github.com/gojek/turing-experiments/runs/7515034880?check_suite_focus=true

I dug deeper into setting a high value for a buffered channel and this seems to create a problem where too much memory is allocated. One thing that I haven't fully figured out though, is why we didn't face issues previously when `makechan` is supposed to throw an [error](https://github.com/golang/go/blob/master/src/runtime/chan.go#L76), i.e `1<<16 = 65536` and `1073741824` clearly exceeds the limit.

Reply:
Ah, it was progressive logging.

So originally, the logs only showed `Detected KafkaLogger...`, then it terminated with what was shown in the screenshot. I added another [log previously](https://github.com/gojek/turing-experiments/pull/26/commits/b6acfd3d70c906782f1b873958ace5ebe8573920#diff-b9a96c0bbd14362b23737368d794bc95f998578dd7a7574c7e54471ed0000ae5L102), which should have shown `Configuring publisher...` should the channel be correctly created, but that logging statement was never shown.

<img width="506" alt="image" src="https://user-images.githubusercontent.com/25025366/180965970-67a08613-6e87-429f-8ab7-6c0a44ce00a8.png">

Reply:
I see.. Ok. Also, shall we trigger multiple test runs (like 5-10 times) to see if there is any more flakiness? If not, it could be safe to assume the buffer size change fixed it and we can merge this PR.

Reply:
👍🏻 Agreed, I'll trigger it a few more times on this PR prior to merging.